hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -223,9 +223,13 @@ public static Schema addMetadataFields(Schema schema, boolean withOperationField
   }
 
   public static Schema removeMetadataFields(Schema schema) {
+    return removeFields(schema, HoodieRecord.HOODIE_META_COLUMNS);
+  }
+
+  public static Schema removeFields(Schema schema, List<String> fieldsToRemove) {
     List<Schema.Field> filteredFields = schema.getFields()
                                               .stream()
-                                              .filter(field -> !HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION.contains(field.name()))
+                                              .filter(field -> !fieldsToRemove.contains(field.name()))
                                               .map(field -> new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultVal()))
```

Comment:
Hey, is this right ? the logic is not equivalent: HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION -> HoodieRecord.HOODIE_META_COLUMNS, i added the HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION because when flink turns on the 'changelog mode', there is another metadata field named '_hoodie_operation'

Reply:
is "_hoodie_operation" considered a meta field? then yeah, we need to fix this.

Reply:
`_hoodie_operation` is a metadata field !

Reply:
Thanks @danny0405 for pointing this out! I'll create a patch to fix this today.